spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,159 @@
+      get :index, params: {}, session: valid_session
+      expect(assigns(:posts)).to eq([post])
+    end
+  end
+
+  describe "GET #show" do
+    it "assigns the requested post as @post" do
+      post = Post.create! valid_attributes
+      get :show, params: {id: post.to_param}, session: valid_session
+      expect(assigns(:post)).to eq(post)
+    end
+  end
+
+  describe "GET #new" do
+    it "assigns a new post as @post" do
+      get :new, params: {}, session: valid_session
+      expect(assigns(:post)).to be_a_new(Post)
+    end
+  end
+
+  describe "GET #edit" do
+    it "assigns the requested post as @post" do
+      post = Post.create! valid_attributes
+      get :edit, params: {id: post.to_param}, session: valid_session
+      expect(assigns(:post)).to eq(post)
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new Post" do
+        expect {
+          post :create, params: {post: valid_attributes}, session: valid_session
+        }.to change(Post, :count).by(1)
+      end
+
+      it "assigns a newly created post as @post" do
+        post :create, params: {post: valid_attributes}, session: valid_session
+        expect(assigns(:post)).to be_a(Post)
+        expect(assigns(:post)).to be_persisted
+      end
+
+      it "redirects to the created post" do
+        post :create, params: {post: valid_attributes}, session: valid_session
+        expect(response).to redirect_to(Post.last)
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns a newly created but unsaved post as @post" do
+        post :create, params: {post: invalid_attributes}, session: valid_session
+        expect(assigns(:post)).to be_a_new(Post)
+      end
+
+      it "re-renders the 'new' template" do
+        post :create, params: {post: invalid_attributes}, session: valid_session
+        expect(response).to render_template("new")
+      end
+    end
+  end
+
+  describe "PUT #update" do
+    context "with valid params" do
+      let(:new_attributes) {
+        skip("Add a hash of attributes valid for your model")
+      }
+
+      it "updates the requested post" do
+        post = Post.create! valid_attributes
+        put :update, params: {id: post.to_param, post: new_attributes}, session: valid_session
+        post.reload
+        skip("Add assertions for updated state")
+      end
+
+      it "assigns the requested post as @post" do
+        post = Post.create! valid_attributes
+        put :update, params: {id: post.to_param, post: valid_attributes}, session: valid_session
+        expect(assigns(:post)).to eq(post)
+      end
+
+      it "redirects to the post" do
+        post = Post.create! valid_attributes
+        put :update, params: {id: post.to_param, post: valid_attributes}, session: valid_session
+        expect(response).to redirect_to(post)
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns the post as @post" do
```

Comment:
Similar code found in posts_controller_spec.rb:116, posts_controller_spec.rb:130 (mass = 52)